### DISCUSSION TITLE
Calculate the slip system rotation axis with the slip plane and slip direction

### DISCUSSION THREAD

**Discussion by saberrider24** [mtex-toolbox/mtex#1591] [Ask Anything]
Hello MTEX community,

I have a simple quesiton regurding the calculation of the rotation axis angle in the crystal coordinate system. The reason that I want to make such calculation is to reverse it, so that I can calculate the slip plane in the crystal coordinate system with the rotation axis and slip direction. So that if I have a rotation axis in the sample coordinate system, I can calculate the slip plane trace in the sample coordinate system.  Maybe there is a better and simpler way to do it, but I can only think of such method at the moment. 

What I am planing to do it to get the vector of the rotation axis. Take FCC material as an example, so the rotation axis should be  <11-2>. And I have used the slipSystem commend to compute the slip system. 
```
%consider FCC slip
sSFCC=slipSystem.fcc(ebsd.CS);
% and all symmetrically equivalent variants
sSFCC = sSFCC.symmetrise;
```
Now that the sSFCC containes variables 'b' for slip direction and 'n' for slip plane normal, both have 24x1 Miller value. Since the rotation axis is othogonal to both the slip direction and plane normal,  just make a cross product of them and I can get the rotation angle axis. So I used this simple commend. 
```
RA=cross(sSFCC.n,sSFCC.b)

RA = Miller (show methods, plot)
 size: 24 x 1
 mineral: Austenite (432)
        u       v       w
   0.2778  0.5556  0.2778
  -0.2778  0.2778  0.5556
```
But I am not sure what information is contained in the variable 'RA' now, therefore I am not sure how to proceed with it. It would be appriciated if you can help me solve this problem, or offer any advice on some other methods to calculate the slip plane from the rotation axis. 

Best regards

**Comment by Tijmenvermeij** [comment]
I don't understand the problem, if I use your piece of code, I get 24 Miller variables that seem to be correct:

```
sS = slipSystem.fcc(crystalSymmetry('432'));
sS=sS.symmetrise;
RA = cross(sS.n,sS.b)

RA = Miller (show methods, plot)
 size: 24 x 1
 symmetry: 432
   u  v  w
   1 -1 -2
  -1 -2 -1
  -2 -1  1
   2  1 -1
  -1  1  2
   1  2  1
  -1  1 -2
   1  2 -1
  -2 -1 -1
   2  1  1
  -1 -2  1
   1 -1  2
  -1  2 -1
   1  1 -2
  -2  1  1
   2 -1 -1
  -1 -1  2
   1 -2  1
   1 -2 -1
  -1 -1 -2
  -2  1 -1
   2 -1  1
  -1  2  1
   1  1  2
```

**Reply by saberrider24** [reply]
Thank you for the answer. I think the code is working as I expected and it should contain the UVM and HKL variables.  But I want to reverse the calculation. I can calculate the rotation axis in a data set, and I want to calculate the slip plane based on the rotation axis. However, I do not know which commend should I use. It would be great if you can give me some suggestion.

**Comment by kilir** [comment]
Hi,
so if I understand you correctly, you are asking whether it is possible to get a plane (n) from a single axis (r) which is inside the plane? (b x n = r) 
If you know b, yes. If you don't, I would rather  say no.

Cheers,
Rüdiger

**Reply by saberrider24** [reply]
Dear Rüdiger,

Yes, that is what I wanted to do. The b (slip direction) is fixed for the given silp system, however it normally is a group of directions. My plane is to calculate the possible slip plane normal n with a given rotation axis r and the slip directions b. But I don't know what commend or function should I use for such calculation. Would you pleave give me some suggestions? 

Best regards

**Reply by Tijmenvermeij** [reply]
I guess you should figure out how to rewrite the cross product to get n, and then use the proper Mtex commands? Maybe google knows...

**Reply by ralfHielscher** [reply]
what about
```
n = cross(r,b)
```
this gives you the normal direction from the rotation axis and the slip direction.
Ralf.
